src/actions/SetMeshFESpaceAction.C

```diff
@@ -0,0 +1,36 @@
+#include "SetMeshFESpaceAction.h"
+
+registerMooseAction("PlatypusApp", SetMeshFESpaceAction, "set_mesh_fe_space");
+
+InputParameters
+SetMeshFESpaceAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Set the mesh nodal finite element space to the same as the mesh "
+                             "displacement variable, if one is specified.");
+  return params;
+}
+
+SetMeshFESpaceAction::SetMeshFESpaceAction(const InputParameters & parameters) : Action(parameters)
+{
+}
+
+void
+SetMeshFESpaceAction::act()
+{
+  auto * mfem_problem = dynamic_cast<MFEMProblem *>(_problem.get());
+
+  if (!mfem_problem)
```

Comment:
Is this check required? It prevented segaults in the unused auxkernel and preprocessor tests but now they've been removed. Also checked in:
		- AddVectorCoefficientAction
		- AddProblemOperatorAction
		- AddFESpaceAction
		- AddCoefficientAction

Reply:
I think so - Platypus should still work on regular MOOSE problems, and it's entirely possible a user could accidentally leave MFEM-specific types in the problem without using `MFEMProblem`, in which case we'd prefer a graceful failure rather than a segfault. We could definitely improve the warnings/failure behaviour in this case though - probably a topic for a future PR